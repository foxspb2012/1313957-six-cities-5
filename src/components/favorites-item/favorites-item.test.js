import React from 'react';
import history from '../../history';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import FavoritesItem from './favorites-item';
import configureStore from 'redux-mock-store';

const NameSpace = {
  APP: `APP`,
  USER: `USER`,
  OFFERS: `OFFERS`,
  REVIEWS: `REVIEWS`,
};

const getOffersMock = (count) => {
  const templateOffers = Array(count)
    .fill(``)
    .map((_, index) => {
      return {
        'id': index + 1,
        'city': {
          'name': `Paris`,
          'location': {
            'coordinates': [48.85661, 2.351499],
            'zoom': 13
          }
        },
        'type': `room`,
        'name': `Wood and stone place`,
        'description': `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
        'photos': [`img/apartment-01.jpg`, `img/apartment-03.jpg`],
        'isFavorite': Boolean(Math.random()),
        'isPremium': Boolean(Math.random()),
        'rating': 1,
        'price': 100,
        'bedroomAmount': 1,
        'guestAmount': 1,
        'features': [`Dishwasher`, `Baby seat`],
        'host': {
          'name': `Host`,
          'avatar': `img/avatar-max.jpg`,
          'isPro': false,
          'id': index + 1
        },
        'location': {
          'coordinates': [48.87961000000001, 2.353499],
          'zoom': 16
        },
        'mainPhoto': `img/apartment-03.jpg`,
      };
    });

  if (count === 1) {
    const [oneOffer] = templateOffers;
    return oneOffer;
  }

  return templateOffers;
};

const mockState = {
  [NameSpace.APP]: {
    activeCity: `Paris`,
    activeSortType: `popular`,
    activeOffer: null
  },
  [NameSpace.OFFERS]: {
    offers: [],
    offer: {},
    nearOffers: [],
    favoriteOffers: [],
    error: null,
    isLoaded: false,
    isLoadedOffer: false,
    isLoadedNearOffers: false,
    isLoadedFavoritesOffers: false,
  },
  [NameSpace.REVIEWS]: {
    reviews: [],
    isReviewSending: false,
    error: null
  },
  [NameSpace.USER]: {
    authorizationStatus: `UNAUTHORIZED`,
    email: null
  }
};

describe(`Should FavoritesItem render correctly`, () => {
  it(`Should FavoritesItem render correctly when no offers`, () => {
    const mockStore = configureStore();
    const store = mockStore(mockState);
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <FavoritesItem/>
            </Router>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should FavoritesItem render correctly with offers`, () => {
    const mockStore = configureStore();
    const store = mockStore(mockState);
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <FavoritesItem
                СЃityOffers={getOffersMock(5)}
                city={`Paris`}
                isLoadedFavoritesOffers={true}
              />
            </Router>
          </Provider>

      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
