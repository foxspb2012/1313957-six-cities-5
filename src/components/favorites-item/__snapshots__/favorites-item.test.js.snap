// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoritesItem render correctly Should FavoritesItem render correctly when no offers 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="#"
      >
        <span />
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "fontSize": "38px",
          "fontStyle": "oblique",
          "fontWeight": 700,
          "justifyContent": "flex-start",
          "lineHeight": 0.7,
          "margin": "20px",
          "padding": "30px",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      <h3
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <img
          alt="logo"
          height="50"
          src="/img/logo.svg"
          style={
            Object {
              "marginRight": "20px",
            }
          }
          width="100"
        />
        <svg
          height="100%"
          version="1.1"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            id="path"
          >
            <animate
              attributeName="d"
              begin="0s"
              dur="3s"
              from="m0,110 h0"
              repeatCount="indefinite"
              to="m0,110 h1100"
            />
          </path>
          <text
            fill="black"
          >
            <textPath
              xlinkHref="#path"
            >
              Loading...
            </textPath>
          </text>
        </svg>
      </h3>
    </div>
  </div>
</li>
`;

exports[`Should FavoritesItem render correctly Should FavoritesItem render correctly with offers 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <div
      className="favorites__places"
    >
      <article
        className="place-card favorites__card"
        onMouseEnter={[Function]}
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="place-card__image-wrapper favorites__image-wrapper"
        >
          <a
            href="/offer/1"
            onClick={[Function]}
          >
            <img
              alt="Wood and stone place"
              className="place-card__image"
              height={110}
              src="img/apartment-01.jpg"
              width={150}
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                100
              </b>
              <span
                className="place-card__price-text"
              >
                 night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height={19}
                width={18}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                In bookmarks
              </span>
            </button>
          </div>
          <div
            className="rating__stars place-card__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/1"
              onClick={[Function]}
            >
              Wood and stone place
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Room
          </p>
        </div>
      </article>
      <article
        className="place-card favorites__card"
        onMouseEnter={[Function]}
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="place-card__image-wrapper favorites__image-wrapper"
        >
          <a
            href="/offer/2"
            onClick={[Function]}
          >
            <img
              alt="Wood and stone place"
              className="place-card__image"
              height={110}
              src="img/apartment-01.jpg"
              width={150}
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                100
              </b>
              <span
                className="place-card__price-text"
              >
                 night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height={19}
                width={18}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                In bookmarks
              </span>
            </button>
          </div>
          <div
            className="rating__stars place-card__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/2"
              onClick={[Function]}
            >
              Wood and stone place
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Room
          </p>
        </div>
      </article>
      <article
        className="place-card favorites__card"
        onMouseEnter={[Function]}
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="place-card__image-wrapper favorites__image-wrapper"
        >
          <a
            href="/offer/3"
            onClick={[Function]}
          >
            <img
              alt="Wood and stone place"
              className="place-card__image"
              height={110}
              src="img/apartment-01.jpg"
              width={150}
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                100
              </b>
              <span
                className="place-card__price-text"
              >
                 night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height={19}
                width={18}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                In bookmarks
              </span>
            </button>
          </div>
          <div
            className="rating__stars place-card__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/3"
              onClick={[Function]}
            >
              Wood and stone place
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Room
          </p>
        </div>
      </article>
      <article
        className="place-card favorites__card"
        onMouseEnter={[Function]}
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="place-card__image-wrapper favorites__image-wrapper"
        >
          <a
            href="/offer/4"
            onClick={[Function]}
          >
            <img
              alt="Wood and stone place"
              className="place-card__image"
              height={110}
              src="img/apartment-01.jpg"
              width={150}
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                100
              </b>
              <span
                className="place-card__price-text"
              >
                 night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height={19}
                width={18}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                In bookmarks
              </span>
            </button>
          </div>
          <div
            className="rating__stars place-card__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/4"
              onClick={[Function]}
            >
              Wood and stone place
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Room
          </p>
        </div>
      </article>
      <article
        className="place-card favorites__card"
        onMouseEnter={[Function]}
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="place-card__image-wrapper favorites__image-wrapper"
        >
          <a
            href="/offer/5"
            onClick={[Function]}
          >
            <img
              alt="Wood and stone place"
              className="place-card__image"
              height={110}
              src="img/apartment-01.jpg"
              width={150}
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                100
              </b>
              <span
                className="place-card__price-text"
              >
                 night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height={19}
                width={18}
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                In bookmarks
              </span>
            </button>
          </div>
          <div
            className="rating__stars place-card__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/5"
              onClick={[Function]}
            >
              Wood and stone place
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            Room
          </p>
        </div>
      </article>
    </div>
  </div>
</li>
`;
